src/modules/Builder/hooks/useActivitiesRedirection.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useParams, useNavigate, generatePath } from 'react-router-dom';
+import debounce from 'lodash.debounce';
+
+import { page } from 'resources';
+import { applet } from 'shared/state';
+
+import { useCurrentActivity } from './useCurrentActivity';
+
+export const useActivitiesRedirection = () => {
+  const { appletId } = useParams();
+  const { activity } = useCurrentActivity();
+  const navigate = useNavigate();
+
+  const loadingStatus = applet.useResponseStatus() ?? {};
+
+  useEffect(() => {
+    if (loadingStatus !== 'loading' && !activity) {
+      debounce(() => navigate(generatePath(page.builderAppletActivities, { appletId })));
```

Comment:
what do you need "debounce" here?

Reply:
to put redirection after all synced actions in js stack (sometimes redirect can be called before the data in the session storage updated)

Reply:
pls left a comment for it

Reply:
done